orchestra/communication/mail.py

```diff
@@ -0,0 +1,44 @@
+from django.core.mail import send_mail as _send_mail
+from orchestra.models import CommunicationPreference
+from orchestra.models import Worker
+
+
+def send_mail(subject, message, from_email,
+              recipient_list, fail_silently=False,
+              auth_user=None, auth_password=None,
+              connection=None, html_message=None,
+              communication_type=None):
+    """
+        Light wrapper over Django's send_mail which filters out recipients who
+        opted out of using email via their CommuncationPreference object.
+    """
+    if communication_type is not None:
+        recipient_list = list(filter(lambda email: _filter_recipient(
```

Comment:
[email for communication_type, email in recipient_list if _can_email(communication_type, recipient)] is more pythonic, with _can_email being like `filter_recipient` but returning True/False